src/syrupy/extensions/amber/serializer.py

```diff
@@ -88,24 +88,14 @@ class Marker:
         Name = "name"
         Divider = "---"
 
-    @classmethod
-    def __maybe_int(cls, part: str) -> Tuple[int, Union[str, int]]:
-        try:
-            # cast to int only if the string is the exact representation of the int
-            # for example, '012' != str(int('012'))
-            i = int(part)
-            if str(i) == part:
-                return (1, i)
-            return (0, part)
-        except ValueError:
-            # the nested tuple is to prevent comparing a str to an int
-            return (0, part)
-
     @classmethod
     def __snapshot_sort_key(
         cls, snapshot: "Snapshot"
     ) -> List[Tuple[int, Union[str, int]]]:
-        return [cls.__maybe_int(part) for part in snapshot.name.split(".")]
+        if "." in snapshot.name:
+            index = snapshot.name.rfind(".") + 1
+            return f"{snapshot.name[:index-1]}.{snapshot.name[index:].zfill(4)}"
```

Comment:
The zfill is arbitrary but should ensure sorting of at least 1000 assertions within a single test case.